src/Compilers/CSharp/Portable/SymbolDisplay/ObjectDisplay.cs

```diff
@@ -483,33 +483,38 @@ internal static string FormatLiteral(ulong value, ObjectDisplayOptions options)
             return pooledBuilder.ToStringAndFree();
         }
 
-        internal static string FormatLiteral(double value, ObjectDisplayOptions options)
+        internal static string FormatLiteral(double value, ObjectDisplayOptions options, CultureInfo cultureInfo = null)
         {
             ValidateOptions(options);
 
-            var result = value.ToString("R", CultureInfo.InvariantCulture);
+            var result = value.ToString("R", GetFormatCulture(cultureInfo));
 
             return options.IncludesOption(ObjectDisplayOptions.IncludeTypeSuffix) ? result + "D" : result;
         }
 
-        internal static string FormatLiteral(float value, ObjectDisplayOptions options)
+        internal static string FormatLiteral(float value, ObjectDisplayOptions options, CultureInfo cultureInfo = null)
         {
             ValidateOptions(options);
 
-            var result = value.ToString("R", CultureInfo.InvariantCulture);
+            var result = value.ToString("R", GetFormatCulture(cultureInfo));
 
             return options.IncludesOption(ObjectDisplayOptions.IncludeTypeSuffix) ? result + "F" : result;
         }
 
-        internal static string FormatLiteral(decimal value, ObjectDisplayOptions options)
+        internal static string FormatLiteral(decimal value, ObjectDisplayOptions options, CultureInfo cultureInfo = null)
```

Comment:
Default value must be a compile time constant which `CultureInfo.InvariantCulture` is not.
The alternatives would thus be to inline all cals to `GetFormatCulture` or overload all `FormatLiteral` methods.